orttraining/orttraining/test/python/orttraining_test_checkpoint_aggregation.py

```diff
@@ -0,0 +1,128 @@
+import copy
+from functools import partial
+import inspect
+import math
+from numpy.testing import assert_allclose
+import onnx
+import os
+import pytest
+import torch
+import subprocess
+
+import onnxruntime
+from onnxruntime.capi.ort_trainer import IODescription as Legacy_IODescription,\
+                                         ModelDescription as Legacy_ModelDescription,\
+                                         LossScaler as Legacy_LossScaler,\
+                                         ORTTrainer as Legacy_ORTTrainer
+from onnxruntime.training import _utils, amp, checkpoint, optim, orttrainer, TrainStepInfo,\
+                                      model_desc_validation as md_val,\
+                                      orttrainer_options as orttrainer_options
+
+import _test_commons, _test_helpers
+
+from orttrainer_bert_toy_onnx import generate_random_input_from_model_desc,\
+                                     bert_model_description, load_bert_onnx_model
+
+
+# Test ZeRO checkpoint loading from a distributed Zero stage 1 mixedprecision run to the
+# following configs. This test ensures that the weights as well as the optimizer
+# state are aggregated and loaded correctly into fp32 and fp16 single-node
+# trainers respectively, for Adam and Lamb optimizers.
+@pytest.mark.parametrize("optimizer, mixedprecision_enabled, expected_eval_loss", [
+    (optim.LambConfig(), False, [11.011026]),
+    (optim.AdamConfig(), False, [10.998348]),
+    (optim.LambConfig(), True, [11.011026]),
+    (optim.AdamConfig(), True, [10.998348]),
+])
+def testToyBertCheckpointLoadZero(optimizer, mixedprecision_enabled, expected_eval_loss):
+    # Common setup
+    rtol = 1e-03
```

Comment:
We have been using atol=1e-7 for accuracy. rtol=1e-3 doesn't capture a lot of errors, especially for fp32 scenarios